src/OpenTelemetry.Extensions.Hosting/OpenTelemetryServicesExtensions.cs

```diff
@@ -47,8 +46,10 @@ public static OpenTelemetryBuilder AddOpenTelemetry(this IServiceCollection serv
     {
         Guard.ThrowIfNull(services);
 
-        services.TryAddEnumerable(
-            ServiceDescriptor.Singleton<IHostedService, TelemetryHostedService>());
+        if (!services.Any((ServiceDescriptor d) => d.ServiceType == typeof(IHostedService) && d.ImplementationType == typeof(TelemetryHostedService)))
```

Comment:
Not able to comment on the line but should the summary/remarks this method calls this out?

Reply:
Good idea to update the remarks! I would make it "Notes:" and then use a bulleted list. There should be other examples around doing that. Probably also a good idea to update the CHANGELOG. Something like `* Changed the behavior of the OpenTelemetryBuilder.AddOpenTelemetry extension to INSERT OpenTelemetry services at index 0 instead of ADDING at the end to improve the experience of users capturing telemetry in hosted services registered before the OpenTelemetry registration.` Kind of a mouthful feel free to improve it 😄

Reply:
the entire `TelemetryHostedService` is an internal implementation detail, so probably okay to not mention about it in the summary.

Reply:
changelog suggestions : lets clarify that this still does not guarantee as the TelemetryHostedService may not have done initializing while other hosted services start, so possible to still miss telemetry until its fully initialized.

Reply:
Let me know if you think the wording needs any additional clarity.

Reply:
> lets clarify that this still does not guarantee as the TelemetryHostedService may not have done initializing while other hosted services start, so possible to still miss telemetry until its fully initialized

I didn't dig into older versions but the current code seems to [await each IHostedService.StartAsync one-by-one](https://github.com/dotnet/aspnetcore/blob/2335aa56f519790c4b4b53b955b26f29f7030d36/src/Hosting/Hosting/src/Internal/HostedServiceExecutor.cs#L38) so this new logic of inserting at 0 should actually work really well. The only case where it wouldn't work (that I can think of) is if user inserted their service at index 0 after we stuck our service there 😄

Reply:
got it. The readme can callout this as well. (that the only time we'll miss telemetry from a hosted service is if user manually inserts its ahead of ours)

Reply:
Note, that in .NET 8 hosted service can be configured to start [concurrently](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.hostoptions.servicesstartconcurrently?view=dotnet-plat-ext-8.0&viewFallbackFrom=net-8.0#microsoft-extensions-hosting-hostoptions-servicesstartconcurrently) with `HostOptions.ServicesStartConcurrently = true`